src/hevm/src/EVM/Format.hs

```diff
@@ -336,48 +335,154 @@ prettyvmresult (EVM.VMSuccess (ConcreteBuffer msg)) =
 prettyvmresult (EVM.VMSuccess (SymbolicBuffer msg)) =
   "Return: " <> show (length msg) <> " symbolic bytes"
 
+
+currentSolc :: DappInfo -> VM -> Maybe SolcContract
+currentSolc dapp vm =
+  let
+    this = vm ^?! EVM.env . EVM.contracts . ix (view (EVM.state . EVM.contract) vm)
+    h = view EVM.codehash this
+  in
+    preview (dappSolcByHash . ix h . _2) dapp
+
 -- TODO: display in an 'act' format
-data BranchData = BranchData {
-  _navigation :: String,
-  _constraint :: String,
-  _leafData :: String
+--
+-- TreeLine describes a singe line of the tree
+-- it contains the indentation which is prefixed to it
+-- and its content which contains the rest
+data TreeLine = TreeLine {
+  _indent   :: String,
```

Comment:
will this always just be a bunch of spaces? Maybe int is a better type for it then?

Reply:
no, its the line segment of the tree. e.g. it will be ` |    | |      | `

Reply:
immagine different symbols with different spacing
stupid github markdown screws it up

Reply:
I see. Do you think you have reimplemented [showTree](https://hackage.haskell.org/package/tree-view-0.5/docs/Data-Tree-View.html) here?